text/0000-comprehensive-changelogs.md

```diff
@@ -0,0 +1,145 @@
+- Start Date: 2019-02-04
+- RFC PR: (leave this empty)
+- Gatsby Issue: (leave this empty)
+
+# Summary
+
+The aim of this RFC is to make Gatsby's changelogs more _comprehensive_, by reducing the number of changes that don't get added to changelogs.
+
+Note: this RFC is specifically not about changing Gatsby's release process to semver (or something else). That's a discussion for another RFC.
+
+# Motivation
+
+A changelog is a markdown file listing all the versions of a package. For each version there will be a list of entries, with one _changelog entry_ describing one change to the package. [Example changelog](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/CHANGELOG.md).
+
+For Gatsby, a useful changelog has the following qualities:
+
+- _comprehensive_ - every pull request merged into the Gatsby repo should create a changelog entry in the relevant package
+- _clear_ - each changelog entry should have a description about the change and what impact it has
+- _concise_ - each changelog entry should be a maximum of one or two paragraphs
+- _link to further information_ - each changelog entry should link to the related pull request, credit the author, and where necessary link to any further documentation
+
+Providing useful changelogs means we have a human-readable and accurate summary of changes over time. This is useful for:
+
+- developers using Gatsby - good changelogs give confidence when updating to newer versions of Gatsby (or official Gatsby packages)
+- developers considering Gatsby for a project - good changelogs are often used as a quality indicator for open source projects
+- Gatsby Inc - good changelogs serve as a public reference, making it easier to highlight new features to Gatsby users (both current and future)
+
+The purpose of this RFC is to discuss making Gatsby's changelogs more _comprehensive_.
+
+There will be a separate RFC for updating the formatting of Gatsby's changelogs, which will discuss making the changelogs more _clear_, _concise_, and _linking to further information_.
+
+# Detailed design
+
+## Ensure changelogs are comprehensive
+
+Currently we generate changelogs based on [conventional commit style](https://www.conventionalcommits.org/en/v1.0.0-beta.3/) PR titles. This is good where it works but prone to human error, as GitHub doesn't provide a way to enforce the format of a PR title.
+
+A pull request with a title like _chore(gatsby): Correct fieldOwners in joi schema_, will be added to the changelog for the `gatsby` package. Changing that title to remove the conventional commit prefix, _Correct fieldOwners in joi schema_, would result in the change not being added to any changelog.
```

Comment:
This is not quite accurate - right now changelog entries are only added when using feat, fix or perf prefixes. That's because we use default changelog generation preset used by lerna (or deps of lerna - didn't research that much), which is `conventional-changelog-angular` (note about type in preset docs: https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular#type and relavant part in code of preset: https://github.com/conventional-changelog/conventional-changelog/blob/d3c2432f695d1b7f33c384be0bcb6495274ddca8/packages/conventional-changelog-angular/writer-opts.js#L44-L45 )

We definitely can fork this preset and adjust to our needs. I did some local tests with adjusting preset to use all the prefixes (result can be seen in https://github.com/pieh/lerna-playground/blob/master/packages/package-b/CHANGELOG.md - with my changes there are actually entries for other prefixes like `docs` or `refactor`)

Reply:
Oops, thanks. Let me update this to make it less wrong 😅

Reply:
I don't think forking is big problem here. We would need to do it (or change preset at very least) if we want to enable changelog entries for different commit types and extra notes, because right now it will pull text for extra config only if `BREAKING CHANGE` is used (which we obviously don't want to use for something that is not breaking change. 

Scope doesn't actually tells which changelog should be generated - it only adds this part to changelog
![screenshot 2019-02-04 at 16 02 26](https://user-images.githubusercontent.com/419821/52216360-54894b00-2896-11e9-8a0d-c4f96b3a200d.png)
Lerna is smart enough to check what packages were changed in commit and adds changelog entry for each of those packages.

Reply:
oh yeah we have have a changelog in each package.. My bad I was thinking of a global one.

so actually the scope doesn't has to be the package name as lerna figures it out. So we just need consistent commits so we can tell lerna what types we support?